core/capabilities/remote/trigger_subscriber.go

```diff
@@ -231,14 +235,30 @@ func (s *triggerSubscriber) Receive(_ context.Context, msg *types.MessageBody) {
 					continue
 				}
 				s.lggr.Infow("remote trigger event aggregated", "triggerEventID", meta.TriggerEventId, "capabilityId", s.capInfo.ID, "workflowId", workflowID)
-				registration.callback <- aggregatedResponse
+				s.sendResponse(workflowID, aggregatedResponse)
 			}
 		}
 	} else {
 		s.lggr.Errorw("received trigger event with unknown method", "method", SanitizeLogString(msg.Method), "sender", sender)
 	}
 }
 
+func (s *triggerSubscriber) sendResponse(workflowID string, response commoncap.TriggerResponse) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	registration, found := s.registeredWorkflows[workflowID]
+	if found {
+		select {
+		case registration.callback <- response:
+		default:
+			s.lggr.Warn("slow consumer detected, closing subscription", "capabilityId", s.capInfo.ID, "workflowId", workflowID)
+			s.closeSubscription(workflowID)
```

Comment:
I don't think we should close it. We're not giving the engine a chance to recover. I admit that with a channel size of 1000 it's pretty unlikely for it to recover. But still, maybe some large buffering happens in the network somewhere and we'll get a flood of past requests. I'd rather not get into a state that is not recoverable, apart from a node restart. How about we just log an error here?

Reply:
you mean log an error and drop the response?  Thinking purely in terms of current use cases, logging and dropping the response is the better solution, so that may be the way to go.  My concern was that you may see some quite hard to diagnose unexpected behaviours further downstream if you intermittently drop responses in different non-price feed contexts, and in this scenario it might be cleaner to break the subscription.  Ideally, if it was aware of the type of data it could conflate in the case of price feeds, but at this level the code is data type agnostic.   

We should probably stat the % filled of the buffer to metrics and monitor, it would be a good indicator that the system is running too hot, I'll add that in.

Reply:
Yes definitely let's add a metric for % filled and a loud error when dropping.

I don't think it makes sense to end up in a state that requires a node restart. That's why I don't want to cancel a sub. Firing on a subset of trigger events doesn't seem worse than not firing at all. Especially in cases where this problem affects a single node, not all of them at once.